Riot/Modules/SetPinCode/SetupBiometrics/SetupBiometricsViewModel.swift

```diff
@@ -76,48 +78,36 @@ final class SetupBiometricsViewModel: SetupBiometricsViewModelType {
     // MARK: - Private
     
     private func enableDisableBiometrics() {
-        LocalAuthenticationService.isShowingBiometrics = true
-        LAContext().evaluatePolicy(.deviceOwnerAuthentication, localizedReason: VectorL10n.biometricsUsageReason) { (success, error) in
-            if success {
-                //  complete after a little delay
-                DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
-                    self.pinCodePreferences.canUseBiometricsToUnlock = nil
-                    self.pinCodePreferences.resetCounters()
-                    self.coordinatorDelegate?.setupBiometricsViewModelDidComplete(self)
-                    LocalAuthenticationService.isShowingBiometrics = false
-                }
-            } else {
-                LocalAuthenticationService.isShowingBiometrics = false
+        biometricsAuthenticationPresenter.present(with: VectorL10n.biometricsUsageReason) { (response) in
+            switch response {
+            case .success:
+                self.pinCodePreferences.canUseBiometricsToUnlock = nil
+                self.pinCodePreferences.resetCounters()
+                self.coordinatorDelegate?.setupBiometricsViewModelDidComplete(self)
+            case .failure:
+                break
             }
         }
     }
     
     private func unlockWithBiometrics() {
-        LocalAuthenticationService.isShowingBiometrics = true
-        LAContext().evaluatePolicy(.deviceOwnerAuthentication, localizedReason: VectorL10n.biometricsUsageReason) { (success, error) in
-            if success {
-                //  complete after a little delay
-                DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
-                    self.pinCodePreferences.canUseBiometricsToUnlock = nil
-                    self.pinCodePreferences.resetCounters()
-                    self.coordinatorDelegate?.setupBiometricsViewModelDidComplete(self)
-                    LocalAuthenticationService.isShowingBiometrics = false
-                }
-            } else {
+        biometricsAuthenticationPresenter.present(with: VectorL10n.biometricsUsageReason) { (response) in
+            switch response {
+            case .success:
+                self.pinCodePreferences.canUseBiometricsToUnlock = nil
+                self.pinCodePreferences.resetCounters()
+                self.coordinatorDelegate?.setupBiometricsViewModelDidComplete(self)
+            case .failure(let error):
                 guard let error = error as NSError? else {
                     return
                 }
```

Comment:
Error is bridgeable to NSError, you can do:
```suggestion
                let nsError = error as NSError
```

Reply:
Yes, but it's intentional to simplify two lines into one